### PR TITLE
Refactor multiple precision tests and fix some multiple precision issues

### PR DESCRIPTION
**Pull request by abelsiqueira**
Multiple precision is tested through a function now, so it tests an input problem,  instead of a fixed ADNLPModel. This allows the multiple precision tests to be used in other packages. Also run multiple precision tests in more models.

### CODE REVIEW

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this pull request

| Package Name | master | stable |
|--|--|--|
| AmplNLReader | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650650) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650651) | 
| CUTEst | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650652) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650653) | 
| CaNNOLeS | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650654) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650655) | 
| NLPModelsIpopt | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650656) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650657) | 
| NLPModelsJuMP | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650658) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650659) | 
| QuadraticModels | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650660) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650661) | 
| SolverTools | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650662) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638650663) |

File: test/problems/brownden.jl [hunk 1]
```diff
@@ -10,10 +10,10 @@ function brownden_autodiff()
 
   x0 = [25.0; 5.0; -5.0; -1.0]
   f(x) = begin
-    s = 0.0
+    T = eltype(x)
```
Comment (dpo) [inline test/problems/brownden.jl]: x0 sill has eltype Float64 though, and that will make it into the model. Is that a problem?

File: test/problems/hs10.jl [hunk 2]
```diff
@@ -29,20 +29,20 @@ function NLPModels.obj(nlp :: HS10, x :: AbstractVector)
...
   increment!(nlp, :neval_hess)
-  return spzeros(2, 2)
+  return spzeros(T, 2, 2)
 end
 
-function NLPModels.hess(nlp :: HS10, x :: AbstractVector, y :: AbstractVector; obj_weight=1.0)
+function NLPModels.hess(nlp :: HS10, x :: AbstractVector{T}, y :: AbstractVector; obj_weight=1.0) where T
   increment!(nlp, :neval_hess)
-  return y[1] * [-6.0  0.0; 2.0  -2.0]
+  return y[1] * T[-6.0  0.0; 2.0  -2.0]
```
Comment (dpo) [inline test/problems/hs10.jl]: Should `y` also have eltype `T` here?

File: test/problems/hs11.jl [hunk 3]
```diff
@@ -49,26 +49,26 @@ function NLPModels.hess_structure!(nlp :: HS11, rows :: AbstractVector{Int}, col
...
 
-function NLPModels.hess_coord!(nlp :: HS11, x :: AbstractVector, vals :: AbstractVector; obj_weight=1.0)
+function NLPModels.hess_coord!(nlp :: HS11, x :: AbstractVector{T}, vals :: AbstractVector; obj_weight=one(T)) where T
   increment!(nlp, :neval_hess)
   vals .= 2obj_weight
   return vals
 end
 
-function NLPModels.hess_coord!(nlp :: HS11, x :: AbstractVector, y :: AbstractVector, vals :: AbstractVector; obj_weight=1.0)
+function NLPModels.hess_coord!(nlp :: HS11, x :: AbstractVector{T}, y :: AbstractVector, vals :: AbstractVector; obj_weight=one(T)) where T
```
Comment (dpo) [inline test/problems/hs11.jl]: Here too, should `y` have eltype `T`?

File: test/problems/hs11.jl [hunk 4]
```diff
@@ -49,26 +49,26 @@ function NLPModels.hess_structure!(nlp :: HS11, rows :: AbstractVector{Int}, col
...
 
-function NLPModels.hprod!(nlp :: HS11, x :: AbstractVector, v :: AbstractVector, Hv :: AbstractVector; obj_weight=1.0)
+function NLPModels.hprod!(nlp :: HS11, x :: AbstractVector{T}, v :: AbstractVector, Hv :: AbstractVector; obj_weight=one(T)) where T
   increment!(nlp, :neval_hprod)
   Hv .= 2obj_weight * v
   return Hv
 end
 
-function NLPModels.hprod!(nlp :: HS11, x :: AbstractVector, y :: AbstractVector, v :: AbstractVector, Hv :: AbstractVector; obj_weight=1.0)
+function NLPModels.hprod!(nlp :: HS11, x :: AbstractVector{T}, y :: AbstractVector, v :: AbstractVector, Hv :: AbstractVector; obj_weight=one(T)) where T
```
Comment (dpo) [inline test/problems/hs11.jl]: And here.

File: test/problems/brownden.jl (hunk 1 above)
Comment (abelsiqueira) [reply test/problems/brownden.jl]: The default model is `Float64`. The `obj` called with a different `eltype` of `x` will be fine.

File: test/problems/hs11.jl (hunk 4 above)
Comment (abelsiqueira) [reply test/problems/hs11.jl]: I'll add `{T}` to these.

Comment (abelsiqueira) [reply test/problems/hs11.jl]: Done

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this pull request

| Package Name | master | stable |
|--|--|--|
| AmplNLReader | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768679) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768680) | 
| CUTEst | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768681) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768682) | 
| CaNNOLeS | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768683) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768684) | 
| NLPModelsIpopt | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768685) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768686) | 
| NLPModelsJuMP | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768687) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768688) | 
| QuadraticModels | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768689) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768690) | 
| SolverTools | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768691) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/638768692) |

**Comment by abelsiqueira** [comment]
All fixed now.

**Pull request merged**